user-manager-service/src/main/java/com/crm/food/establishment/user/auth/service/impl/JwtServiceImpl.java

```diff
@@ -0,0 +1,187 @@
+    private final RefreshTokenHandlerAdapter refreshTokenHandlerAdapter;
+
+    private final PrivateKey accessTokenPrivateKey;
+    private final JwtParser accessTokenParser;
+    private final AccessTokenHandlerAdapter accessTokenHandlerAdapter;
+
+    private final RedisTemplate<UUID, String> refreshTokenRedisTemplate;
+
+    public JwtServiceImpl(
+            JwtProperties jwtProperties,
+            RedisTemplate<UUID, String> refreshTokenRedisTemplate
+    ) throws NoSuchAlgorithmException, InvalidKeySpecException {
+        KeyFactory rsaKeyFactory = KeyFactory.getInstance("RSA");
+
+        this.jwtProperties = jwtProperties;
+
+        refreshTokenPrivateKey = convertBase64PrivateKeyToJavaPrivateKey(
+                jwtProperties.refreshToken().secretKey(),
+                rsaKeyFactory
+        );
+        PublicKey refreshTokenPublicKey = convertBase64PublicKeyToJavaPublicKey(
+                jwtProperties.refreshToken().publicKey(),
+                rsaKeyFactory
+        );
+        refreshTokenParser = Jwts.parserBuilder()
+                .setSigningKey(refreshTokenPublicKey)
+                .build();
+        refreshTokenHandlerAdapter = new RefreshTokenHandlerAdapter();
+
+        accessTokenPrivateKey = convertBase64PrivateKeyToJavaPrivateKey(
+                jwtProperties.accessToken().secretKey(),
+                rsaKeyFactory
+        );
+        PublicKey accessTokenPublicKey = convertBase64PublicKeyToJavaPublicKey(
+                jwtProperties.accessToken().publicKey(),
+                rsaKeyFactory
+        );
+        accessTokenParser = Jwts.parserBuilder()
+                .setSigningKey(accessTokenPublicKey)
+                .build();
+        accessTokenHandlerAdapter = new AccessTokenHandlerAdapter();
+
+        this.refreshTokenRedisTemplate = refreshTokenRedisTemplate;
+    }
+
+    @Override
+    public String issueAccessToken(User user) {
+        LocalDateTime issuedTime = LocalDateTime.now().withSecond(0).withNano(0);
+        LocalDateTime expirationTime = issuedTime.plusMinutes(
+                jwtProperties.accessToken().expirationTimeInMinutes()
+        );
+
+        String builtAccessToken = Jwts.builder()
+                .setIssuedAt(convertLocalDateTimeToDate(issuedTime))
+                .setExpiration(convertLocalDateTimeToDate(expirationTime))
+                .setSubject(user.getUuid().toString())
+                .claim("role", user.getRole())
+                .signWith(accessTokenPrivateKey, SignatureAlgorithm.RS256)
+                .compact();
+
+        log.info("Issued access token for user: " + user.getEmail());
+        return builtAccessToken;
+    }
+
+    @Override
+    public String issueRefreshToken(User user) {
+        LocalDateTime issuedTime = LocalDateTime.now().withSecond(0).withNano(0);
+        LocalDateTime expirationTime = issuedTime.plusMinutes(
+                jwtProperties.refreshToken().expirationTimeInMinutes()
+        );
```

Comment:
I wanted to separate extraction of 'expirationTimeInMinutes', so it will be visibly highlighted. If you mean put it in one line, first of all it will be long, secondly it is hard to read for me:
<img width="702" alt="image" src="https://github.com/polinaucc/crm-food-establishment/assets/95944057/3c5136ae-310a-4199-b8fe-41fbcd3d6b8f">